### PR TITLE
Draft: #41 Search results

### PR DESCRIPTION
**Pull request by stevedya**
Addresses ticket Monday Ticket 41. This PR handles initial styling for the search results page.
Pagination to be handled on another ticket.

## Desktop
Screen Shot 2022-08-11 at 11 13 04 AM

## Mobile
Screen Shot 2022-08-11 at 11 17 02 AM

### CODE REVIEW

File: wagtailio/static/sass/components/_buttons.scss [hunk 1]
```diff
@@ -4,7 +4,6 @@
     text-decoration: none;
     border-radius: 6px;
     border: 1px solid $color--off-black;
-    margin: 0 auto;
```
Comment (stevedya) [inline wagtailio/static/sass/components/_buttons.scss:7]: Having the buttons always align themselves could cause us extra work removing margins in the future

File: wagtailio/project_styleguide/templates/patterns/components/search.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{% extends "patterns/base_page.html" %}
+{% load wagtailcore_tags %}
+
+{% block content %}
+    <div class="grid">
+        <div class="hero hero--search">
+            <div class="hero__inner">
+                <h1 class="hero__heading">Search results</h1>
+
+                <div>
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/components/search.html:10]: question: i'm wondering if this div can be removed?

File: wagtailio/static/sass/components/_buttons.scss (hunk 1 above)
Comment (chris-lawton) [reply wagtailio/static/sass/components/_buttons.scss:7]: I agree. I think it was added so the buttons display normally [in the pattern library](http://localhost:8000/pattern-library/pattern/patterns/styleguide/buttons.html). We could add an inline `<style>` tag on the [buttons PL template](https://github.com/wagtail/wagtail.org/blob/next/wagtailio/project_styleguide/templates/patterns/styleguide/buttons.html) and move that rule here so they're not full width in the PL

File: wagtailio/static/sass/components/_utilities.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.u {
+  &-sr-only {
```
Comment (chris-lawton) [inline wagtailio/static/sass/components/_utilities.scss:2]: We've added this in `utilities/_utilities.scss` : )

File: wagtailio/project_styleguide/templates/patterns/includes/svg_sprites.html [hunk 4]
```diff
@@ -119,6 +119,10 @@
         <path d="M74.406 78.238l-32.835-4.606-8.919-63.458 32.836 4.606 8.918 63.458zm-3.703-64.526l19.7-11.32c2.327-1.332 5.54.06 5.933 2.855l7.741 55.074c.231 1.644-.548 3.262-2.122 3.987L79.621 77.17 70.703 13.71zM4.94 24.127l22.498-12.885L36.356 74.7 16.491 86.042c-2.327 1.333-5.376-.082-5.769-2.877L2.982 28.09c-.23-1.644.383-3.24 1.958-3.964z" fill="currentColor"/>
     </symbol>
 
+    <symbol id="magnifying-glass" viewBox="0 0 21 22">
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/includes/svg_sprites.html:122]: Suggestion: this [icon exists here](https://github.com/wagtail/wagtail.org/blob/next/wagtailio/project_styleguide/templates/patterns/includes/svg_sprites.html#L114) however the one you've added works better with the `icon.html` template so let's remove the one i've linked to and keep yours.

File: wagtailio/static/sass/abstracts/_variables.scss [hunk 5]
```diff
@@ -25,3 +25,9 @@ $breakpoints: ('medium' '(min-width: 599px)', 'large' '(min-width: 1023px)');
 // Site grid
 $site-width: 1440px;
 $gutter: 30px;
+
+// Border Radius
+$rounded-sm: 3px;
+$rounded-md: 6px;
+$rounded-lg: 10px;
+$rounded-xl: 15px;
```
Comment (chris-lawton) [inline wagtailio/static/sass/abstracts/_variables.scss:33]: thanks for adding these! : )

File: wagtailio/project_styleguide/templates/patterns/components/search.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{% extends "patterns/base_page.html" %}
```
Comment (chris-lawton) [inline wagtailio/project_styleguide/templates/patterns/components/search.html:1]: suggestion: what do you think to naming this file `search-results.html`?

**Comment by chris-lawton** [comment]
Nice work @stevedya! I'm liking the cross icon that appears when you hover the search input that's used to clear it. What do you think of adding that to the main search dropdown [in the header](http://localhost:8000/pattern-library/pattern/patterns/styleguide/header.html)?

image

File: wagtailio/static/sass/components/_buttons.scss (hunk 1 above)
Comment (stevedya) [reply wagtailio/static/sass/components/_buttons.scss:7]: Awesome, that makes sense. I'll do that now :)

File: wagtailio/static/sass/abstracts/_variables.scss (hunk 5 above)
Comment (stevedya) [reply wagtailio/static/sass/abstracts/_variables.scss:33]: Yass no problem! I'll make a follow up PR after that is for the purpose of replacing where these values are used on other templates too!

File: wagtailio/static/sass/components/_utilities.scss (hunk 3 above)
Comment (stevedya) [reply wagtailio/static/sass/components/_utilities.scss]: haha nice one! I've removed this template now

File: wagtailio/project_styleguide/templates/patterns/components/search.html (hunk 2 above)
Comment (stevedya) [reply wagtailio/project_styleguide/templates/patterns/components/search.html]: Yep great catch! I've removed this now.

File: wagtailio/project_styleguide/templates/patterns/components/search.html (hunk 6 above)
Comment (stevedya) [reply wagtailio/project_styleguide/templates/patterns/components/search.html]: Sounds great to me! Totally forgot you had been doing the search component in tandem

**Comment by stevedya** [comment]
@chris-lawton Good shout, I've converted the header search box to a `type=search` so that it will have the same cross now 👍 I've also just gone through the feedback so this ready to go now :)

**Review by chris-lawton — APPROVED** [review]

**Pull request merged**